### ISSUE TITLE
not validate russian urls

### ISSUE THREAD

**Issue by Mikkou** [vlucas/valitron#234]
for example: "http://капоэйра-детям.рф"

**Comment by willemwollebrants** [comment]
This is due to a bug with php's built-in FILTER_VALIDATE_URL I'm afraid.
I'm going to investigate if there is a way around this

**Comment by Rotzbua** [comment]
@willemwollebrants No, FILTER_VALIDATE_URL works correctly. Unicode is not allowed by the rfc specification. Every browser converts the unicode in the background to ascii.

Example:
```php
<?php
var_dump(filter_var('http://example.com', FILTER_VALIDATE_URL));
// xss
var_dump(filter_var('javascript://comment%0Aalert(1)', FILTER_VALIDATE_URL));
// non ascii
var_dump(filter_var('http://www.müller.de', FILTER_VALIDATE_URL));
// unicode not allowed in url must be converted to ascii
//echo idn_to_ascii('http://www.müller.de'); // add. extension must be installed
var_dump(filter_var('http://www.xn--mller-kva.de', FILTER_VALIDATE_URL));
// russian
var_dump(filter_var('http://капоэйра-детям.рф', FILTER_VALIDATE_URL));
var_dump(filter_var('http://xn----7sbarfufwmjn2a0l9a.xn--p1ai', FILTER_VALIDATE_URL));
```

Web converter: https://www.verisign.com/en_US/channel-resources/domain-registry-products/idn/idn-conversion-tool/index.xhtml

It seems that using php build in function is a bad idea: https://d-mueller.de/blog/why-url-validation-with-filter_var-might-not-be-a-good-idea/

**Comment by jabarihunt** [comment]
@Rotzbua To prevent XSS you'd use `FILTER_SANITIZE_URL` rather than `FILTER_VALIDATE_URL`.  This library is for validating rather than sanitizing, so the latter is probably the more valid filter.

@everyone - A possible solution would be to search the domain for non-ascii (unicode) characters.  If none exist, proceed as normal.  If they do, convert the url to all ascii chars first and then proceed.

**Comment by vlucas** [comment]
This was resolved in #232
